Gracefully handle additional oneof fields in SeriesResponse

pracucci opened the pull request:
* [ ] I added CHANGELOG entry for this change.
* [x] Change is not relevant to the end user.

## Changes

Following up the discussion we had in #2479, in this PR I'm proposing to gracefully handle additional `oneof` fields we may add in the future to `SeriesResponse`, without assuming that it either contains warnings or series (may also contain new fields an older client doesn't known).

## Verification

(Already existing)  tests.

bwplotka commented on pkg/store/proxy.go:
```diff
@@ -434,11 +434,15 @@ func startStreamSeriesSet(
+					s.handleErr(errors.Wrapf(ctx.Err(), "failed to receive any data from %s", s.name), done)
+					return
+				}
+				continue
```
do we need continue?

pracucci replied:
No. It's useless as, technically, is now useless the `continue` in the "warnings branch" above. I've removed both for consistency, but if you have a different opinion I can reiterate on it.

pracucci commented:
> I think this is worth a CHANGELOG entry, but let's do on separate PR or during the release as it will create conflict with this #2500

I've rebased `master` and added it. PTAL if you think it's clear enough.

bwplotka commented:
Thanks!

bwplotka commented:
![](https://media0.giphy.com/media/3orifepy4Usb0FVscw/giphy.gif)
